Core/src/org/sleuthkit/autopsy/timeline/events/EventAddedEvent.java

```diff
@@ -18,15 +18,62 @@
  */
 package org.sleuthkit.autopsy.timeline.events;
 
+import java.io.Serializable;
+import java.util.logging.Level;
 import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.coreutils.Logger;
 import org.sleuthkit.autopsy.events.AutopsyEvent;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.timeline.TimelineEvent;
 
 /**
- *
+ * An AutopsyEvent broadcast when a TimelineEvent is added to the case.
  */
-public class EventAddedEvent extends AutopsyEvent {
+public class EventAddedEvent extends AutopsyEvent implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(EventAddedEvent.class.getName());
+
+    private transient TimelineEvent addedEvent;
+    private final long addedEventID;
 
     public EventAddedEvent(org.sleuthkit.datamodel.TimelineManager.EventAddedEvent event) {
-        super(Case.Events.EVENT_ADDED.name(), null, event.getEvent());
+        super(Case.Events.EVENT_ADDED.name(), null, event.getAddedEvent().getEventID());
+        addedEvent = event.getAddedEvent();
+        addedEventID = addedEvent.getEventID();
+    }
```

Comment:
It seems like passing the event id to the super class constructor and also making it a field might be a little confusing. Personally, I prefer the approach used in DataSourceAddedEvent where the id of the object is obtained in getNewValue via long id = (Long) super.getNewValue() and the "extra" field is not required. Thoughts?